### PR TITLE
Fix FTP#list to recurse through sparse dirs, #165

### PR DESCRIPTION
**Pull request by eharik**
* Change `fillBuffer` to ensure that `buffer`
  head has a file (as opposed to file or dir).
* Clean up pattern matching on Vector.

### CODE REVIEW

**Comment by eharik** [comment]
@juanjovazquez looks like you are the original contributor on this component, can you take a look at this PR?

**Comment by juanjovazquez** [comment]
LGTM. Thanks for catching this corner case!.

File: ftp/src/main/scala/akka/stream/alpakka/ftp/impl/FtpBrowserGraphStage.scala [hunk 1]
```diff
@@ -75,14 +62,13 @@ private[ftp] trait FtpBrowserGraphStage[FtpClient, S <: RemoteFileSettings] exte
...
-        buffer match {
-          case Seq() => // Nothing to do
-          case head +: tail =>
-            if (head.isDirectory) {
-              buffer = getFilesFromPath(head.path) ++ tail
-            }
+      private[this] def fillBuffer(): Unit = buffer match {
+        case head +: tail if head.isDirectory => {
+          buffer = getFilesFromPath(head.path) ++ tail
+          fillBuffer()
```
Comment (patriknw) [inline ftp/src/main/scala/akka/stream/alpakka/ftp/impl/FtpBrowserGraphStage.scala]: `@tailrec` ?

Comment (eharik) [inline ftp/src/main/scala/akka/stream/alpakka/ftp/impl/FtpBrowserGraphStage.scala]: Good call, just pushed an update with the annotation.

**Comment by eharik** [comment]
Specs failing, looks to be the same comment as [here](https://github.com/akka/alpakka/pull/167.  I'll rebase and update the timeout.

**Comment by johanandren** [comment]
Refs #165

**Pull request merged**